data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/router/RouterCopyRecordStrategy.java

```diff
@@ -134,4 +134,12 @@ public Collection<Record> getAllRecords(final Collection<Record> allRecords) {
         }
         return newRecords;
     }
+
+    /*
+     * for testing
+     */
+    public AcknowledgementSetManager getAcknowledgementSetManager() {
```

Comment:
Having this just for testing isn't ideal, and normally indicates that there are some places that the code structure could be improved. However, this is the easiest way with how the `RouterCopyRecordStrategy` is instantiated, since you  can't verify calls to a constructor without using PowerMock. If you did want to be able to test without this extra getter method, you could make a factory class for RouterCopyRecordStrategy, and verify the call to that has the correct acknowledgment set manager with an argument captor.

Reply:
Fixed it by removing this and modifying the test case.